Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/SingleFileTransfer.cs

```
@@ -263,80 +295,103 @@ private static Encoding GetEncoding(Destination dest)
                     Client.RenameFile(filePath, SourceFile.FullPath);
+                    _logger.NotifyInformation(BatchContext, $"FILE RENAME: Temporary file {SourceFileDuringTransfer} restored to target {SourceFile.FullPath}.");
                 }
                 break;
         }
     }
 
-    private void CleanUpFiles()
+    private void ExecuteSourceOperationMoveOrRename()
     {
-        if (BatchContext.Options.RenameSourceFileBeforeTransfer && !Path.GetFileName(SourceFileDuringTransfer).Equals(SourceFile.FullPath))
+        var filePath = string.IsNullOrEmpty(SourceFileDuringTransfer) ? SourceFile.FullPath : SourceFileDuringTransfer;
+ 
+        if (BatchContext.Source.Operation == SourceOperation.Move) 
         {
-            Trace(TransferState.CleanUpFiles, "Removing temporary file {0}", SourceFileDuringTransfer);
-            TryToRemoveSourceTempFile(SourceFileDuringTransfer);
+            var moveToPath = _renamingPolicy.ExpandDirectoryForMacros(BatchContext.Source.DirectoryToMoveAfterTransfer);
+            SetCurrentState(TransferState.SourceOperationMove, $"Moving source file {SourceFile.FullPath} to {moveToPath}.");
+            var file = Client.Get(filePath);
+            if (!Client.Exists(moveToPath))
+            {
+                var msg = $"Operation failed: Source file {SourceFile.Name} couldn't be moved to given directory {moveToPath} because it didn't exist.";
+                _logger.NotifyError(BatchContext, msg, new ArgumentException("Failure in moving the source file."));
+                _result.ErrorMessages.Add($"Failure in source operation: {msg}");
+            }
+            var destFileName = moveToPath.Contains("/")
+                ? moveToPath + "/" + SourceFile.Name
+                : Path.Combine(moveToPath, SourceFile.Name);
+
+            file.MoveTo(destFileName);
+            if (!Client.Exists(destFileName)) throw new Exception("");
```

I am not exactly sure what the purpose of this is, could you add a comment to clarify why we need to throw an exception with an empty string here?